nixos/modules/system/boot/initrd-ssh.nix

```diff
@@ -217,6 +229,36 @@ in
 
     boot.initrd.secrets = listToAttrs
       (map (path: nameValuePair (initrdKeyPath path) path) cfg.hostKeys);
+
+    # Systemd initrd stuff
+    boot.initrd.systemd = mkIf config.boot.initrd.systemd.enable {
+      users.sshd = { uid = 1; group = "sshd"; };
```

Comment:
Sidenote: do we assert that these UIDs are unique somewhere?

Reply:
I don't think this does anywhere and it definitely should. Also wish we didn't have to hardcode uid `1` here, but I see why we probably do

Reply:
It doesn't assert that anywhere... FWIW we're hopefully only ever using the implicit `users.foo = {}` style, which inherits IDs from the stage 2 config, which I believe does have such an assertion. `sshd` is the unfortunate exception to this. Anyway, I suppose it's worth asserting uniqueness

Reply:
The assertion was actually extremely easy to add by moving the code for this to the `users-groups.nix` module, which is arguably a more appropriate location anyway.